### DISCUSSION TITLE
Optional async? methods

### DISCUSSION THREAD

**Discussion by omuleanu** [dotnet/csharplang#5907] [Language Ideas]
It would be nice if we could mark `async `methods as "not async" or `sync`, basically to avoid getting the warning:
`CS1998 This async method lacks 'await' operators and will run synchronously.`
This would be useful when setting `Func<Task>` properties or implementing methods that return Task.

example:
```
// setting Func<Task<IEnumerable<T>>> property of an object

GetData = async? () => staticDb.Dinners.ToArray() 
...
GetData = async () => await dinnersQuery.ToArrayAsync()
```
`async?` would mean it's actually sync

The current workaround is to remove the `async` keyword and call `Task.FromResult(res)` (mentioned [here](https://stackoverflow.com/a/13254787/112100) ), 
also `pragma warning disable` and one call to `await Task.Yield()`

**Comment by TonyValenti** [comment]
I blanket disable that warning.

**Reply by jnm2** [reply]
Me too. It puts the pressure in the wrong place, leading to subtle antipatterns. More info under https://github.com/dotnet/csharplang/discussions/5457#discussioncomment-1675435.
